platform/openide.util/test/unit/src/org/openide/util/MapFormatTest.java

```diff
@@ -18,68 +18,257 @@
+
+    }
+
+    //--------------------------------------------------------------------------
+    @Test
+    public void format_acceptsFormatStringWithMissingRightBraces_ifExactMatchIsFalse() {
+
+        MapFormat mf = new MapFormat(this.johnDoe);
+        mf.setExactMatch(false);
+        assertEquals("Hello {prefix Doe.", mf.format("Hello {prefix {last}."));
+        assertEquals("Hello Mr {last.", mf.format("Hello {prefix} {last."));
+    }
+
+    //--------------------------------------------------------------------------
+    @Test
+    public void format_worksWithDatesAndNumbersAsArgumentsAndOther() {
+
+        String result = format("{date} {number}.", this.dateNumberAndClass);
+        // Since exact format depends on default locale we need to verify this 
+        // the result indirectly.
+        System.out.println(result);
+        assertTrue(result.contains("34"));
+        assertTrue(result.contains("8"));
+        assertTrue(result.contains("6"));
+        assertTrue(result.contains("12"));
+        assertTrue(result.contains("23"));
+        assertTrue(result.contains("999"));
+
+        assertEquals("class java.lang.Void", format("{class}", this.dateNumberAndClass));
+    }
+    //--------------------------------------------------------------------------
+    @Test
+    public void parseObject_throwsNullPointer_whenGivenNullArgument() {
+
+        MapFormat mf = new MapFormat(this.johnDoe);
+        
+        try {
+            mf.parseObject(null, new ParsePosition(0));
+            fail();
+        } catch (final NullPointerException e) {
+            //good
+        }
+    }
+    //--------------------------------------------------------------------------
+    @Test
+    public void parseObject_returnsEmptyString_whenGivenEmptyString() {
+        
+        MapFormat mf = new MapFormat(this.johnDoe);
+        
+        assertEquals("", mf.parseObject("", new ParsePosition(0)));
+    }
+    //--------------------------------------------------------------------------
+    @Test
+    public void parseObject_returnsPatternString_whenProperMessageString() {
         
-        HashMap args = new HashMap();
-        args.put("NAME", "Jaroslav");
-
-        MapFormat f = new MapFormat(args);
-        f.setLeftBrace("__");
-        f.setRightBrace("__");
-        f.setExactMatch(false);
-        String result = f.format(s);
+        MapFormat mf = new MapFormat(this.johnDoe);
         
-        assertEquals("Should be ok: " + result, s, result);
+        assertEquals("Hello {first}.", mf.parseObject("Hello John.", new ParsePosition(0)));
+    }
+    //--------------------------------------------------------------------------
+    @Test
+    public void parseObject_returnsItsArgument_whenNoTagsCanBeResolved() {
+        
+        MapFormat mf = new MapFormat(this.johnDoe);
         
+        assertEquals("Hello Adam.", mf.parseObject("Hello Adam.", new ParsePosition(0)));
     }
-    
-    public void testIssue67238() {
```

Comment:
Same here. Knowing that this test is a regression test for issue 67238 is cool. I think we should keep it.